azure-iot-provisioning-devicessdk/samples/register_symmetric_key.py

```diff
@@ -0,0 +1,54 @@
+import os
+import logging
+from azure.iot.provisioning.devicessdk.security.sk_security_provider import (
+    SymmetricKeySecurityProvider,
+)
+from azure.iot.provisioning.devicessdk.device.registration_client_factory import (
+    create_from_security_provider,
+)
+from azure.iot.hub.devicesdk import DeviceClient
+from azure.iot.hub.devicesdk.auth.authentication_provider_factory import from_connection_string
+
+import time
+
+logging.basicConfig(level=logging.INFO)
+
+provisioning_host = os.getenv("PROVISIONING_HOST")
+id_scope = os.getenv("PROVISIONING_IDSCOPE")
+registration_id = os.getenv("PROVISIONING_REGISTRATION_ID")
+symmetric_key = os.getenv("PROVISIONING_SYMMETRIC_KEY")
+
+
+symmetric_key_security_provider = SymmetricKeySecurityProvider(
+    registration_id, symmetric_key, id_scope
+)
+provisioning_client = create_from_security_provider(
+    provisioning_host, symmetric_key_security_provider, "mqtt"
```

Comment:
Is there a semantic reason why the provisioning_host information can't be bundled with the security provider? I'm not saying that I necessarily think that's the right way to do it, but since they both get passed down to the transport, it makes me wonder why provisioning_host isn't part of the security_provider.

Reply:
that's a valid question. i think originally the idea was that we would separate what constitutes "identity" (symmetric key, registration id) from what is specific to the DPS instance itself (Id scope and global device endpoint).
In the node SDK the idScope is passed to the registration client, not the security client. 

@Oliva please check with other SDKs what the design of the parameters for each object constructor is and let's see if there's a common rule everybody abides by. if there is, let's follow it. if there is not, let's figure out what we think is best as a team.

---
In reply to: [265321560](https://github.com/Azure/azure-iot-sdk-python-preview/pull/59#discussion_r265321560) [](ancestors = 265321560)

Reply:
I had already tried both ways(bundling the host inside security provider and keeping it separate) before giving this PR out as the same thing had confused me. The only reason I finally went with accepting 2 parameters (the host and security provider) separately is because  :-

1. the provisioning host is only needed in MQTT for the hostname part. There is no other need for this.
2. The security client on its own does not need this piece of information. It only needs id scope, registration id and the key.

If still we feel that the security client can accept this host and only have a getter defined out of it then I can make the change.

---
In reply to: [265334055](https://github.com/Azure/azure-iot-sdk-python-preview/pull/59#discussion_r265334055) [](ancestors = 265334055,265321560)